src/System/TimedScopes/TimedScopeEventSource.cs

```diff
@@ -0,0 +1,233 @@
+﻿/***************************************************************************************************
+	TimedScopeEventSource.cs
+
+	Copyright (c) Microsoft Corporation.
+
+	Timed scopes event source
+***************************************************************************************************/
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Omex.System.Logging;
+using Microsoft.Omex.System.TimedScopes;
+
+namespace Microsoft.Omex.System.TimedScopes
+{
+	/// <summary>
+	/// Timed scopes event source
+	/// </summary>
+	[EventSource(Name = "Microsoft-OMEX-TimedScopes")]
+	public sealed class TimedScopeEventSource : EventSource
+	{
+		#region Initialization
+
+		/// <summary>
+		/// An initialized instance of the <see cref="TimedScopeEventSource"/> class
+		/// The singleton behavior is needed so that the Event Source works properly
+		/// </summary>
+		public static TimedScopeEventSource Instance => s_instance.Value;
+
+
+		/// <summary>
+		/// Static constructor
+		/// </summary>
+		static TimedScopeEventSource()
+		{
+			// A workaround for the problem where ETW activities do not get tracked until Tasks infrastructure is initialized.
+			// This problem will be fixed in .NET Framework 4.6.2.
+			Task.Run(() => { });
+		}
+
+
+		/// <summary>
+		/// Private constructor to enforce singleton semantics needed for the Event Source to work properly
+		/// </summary>
+		private TimedScopeEventSource()
+			: base()
+		{
+		}
+
+
+		/// <summary>
+		/// Lazy instance of the <see cref="TimedScopeEventSource"/> class
+		/// </summary>
+		private static readonly Lazy<TimedScopeEventSource> s_instance = new Lazy<TimedScopeEventSource>(
+			() => new TimedScopeEventSource(),
+			LazyThreadSafetyMode.PublicationOnly);
+
+		#endregion
+
+
+		#region Non-Events
+
+		/// <summary>
+		/// Logs the occurrence of an activity
+		/// </summary>
+		/// <param name="logCategory">Log category</param>
+		/// <param name="name">TimedScope name</param>
+		/// <param name="subtype">TimedScope subtype</param>
+		/// <param name="metadata">TimedScope metadata</param>
+		/// <param name="userHash">User hash</param>
+		/// <param name="serviceName">Service name</param>
+		/// <param name="result">TimedScope result</param>
+		/// <param name="correlationId">Correlation Id</param>
+		/// <param name="durationMs">TimedScope duration in milliseconds</param>
+		[NonEvent]
+		public void LogEvent(
+				Category logCategory,
+				string name,
+				string subtype,
+				string metadata,
+				string userHash,
+				string serviceName,
+
+				TimedScopeResult result,
+				string correlationId,
+				double durationMs
+			)
```

Comment:
>) [](start = 3, length = 1)

Nitpick: The `)` is normally placed at the end of the last parameter. Applies throughout